kalite/topic_tools/content_models.py

```diff
@@ -0,0 +1,389 @@
+import os
+import json
+
+from django.conf import settings as django_settings
+
+import sqlite3
+
+from peewee import Model, SqliteDatabase, CharField, TextField, BooleanField, ForeignKeyField, PrimaryKeyField, Using, DoesNotExist, fn
+from playhouse.shortcuts import model_to_dict
+
+from .settings import CONTENT_DATABASE_PATH
+from .annotate import update_content_availability
+
+
+# This BaseItem is used to subclass different Models for different languages.
+# This allows us to use a separate database for each language, so that we
+# can reduce performance cost, and keep queries simple for multiple languages.
+# In addition, we can distribute databases separately for each language pack.
+class Item(Model):
+    title = CharField()
+    description = TextField()
+    available = BooleanField()
+    kind = CharField()
+    parent = ForeignKeyField("self", null=True, index=True, related_name="children")
+    id = CharField(index=True)
+    pk = PrimaryKeyField(primary_key=True)
+    slug = CharField()
+    path = CharField(index=True, unique=True)
+    extra_fields = CharField(null=True)
+
+    def __init__(self, *args, **kwargs):
+        kwargs = parse_model_data(kwargs)
+        super(Item, self).__init__(*args, **kwargs)
+
+
+def parse_model_data(item):
+    extra_fields = item.get("extra_fields", {})
+
+    if type(extra_fields) is not dict:
+        extra_fields = json.loads(extra_fields)
+
+    remove_keys = []
+    for key, value in item.iteritems():
+        if key not in Item._meta.fields:
+            extra_fields[key] = value
+            remove_keys.append(key)
```

Comment:
Why not just remove the key directly in this line, rather than adding them to a list?